src/utils/Database/LSMTree/Generators.hs

```diff
@@ -585,7 +585,16 @@ instance SerialiseValue RawBytes where
 deriving newtype instance Arbitrary SerialisedKey
 deriving newtype instance SerialiseKey SerialisedKey
 
-deriving newtype instance Arbitrary SerialisedValue
+instance Arbitrary SerialisedValue where
+  -- some should be large
+  arbitrary = SerialisedValue <$> frequency
+      [ (9, coerce (arbitrary @RawBytes))
+      , (1, coerce (arbitrary @LargeRawBytes))
+      ]
+  shrink (SerialisedValue rb)
+      | RB.size rb > 64 = coerce (shrink (LargeRawBytes rb))
+      | otherwise       = coerce (shrink rb)
```

Comment:
> I think in some parts of the tests I wrote, I assume that SerialisedValues are not large, and I have a special generator for generating values that can be both small and large

Makes sense to unify. I'll take a look.

We could have `SmallSerialisedValue` (0-4k only), although I'm wondering a bit whether that's really needed. The tests still seem to pass?

In the other direction, `LargeBytes` doesn't guarantee that the values are large, they're just much more common. And I don't know if there's a test that strictly needs large values.

>  if large values occur too often, we're not very likely to build up pages of small keys only.

We'll get plenty of pages of small keys only, but true, we'll rarely get the case where a small value will not fit any more. Although having a decent amount of "largish" small values (1k-4k bytes) should help with that. Maybe we should have more of those? Currently we have a bunch of <100 byte values and then just a few above that.

Reply:
> We could have SmallSerialisedValue (0-4k only), although I'm wondering a bit whether that's really needed. The tests still seem to pass?

I was wrong -- I use `UTxOKey` for the `CompactIndex` tests

But good point, maybe we want to change the generation to create larger values. It's not uncommon throughout our code that we have separate cases for small vs. large values, and we probably want a nice distribution of them to use in our tests, so Iike your idea. Some options:
1. `TinySerialisedValue`:  0-100(?) bytes, or some other small number, or depending on `sized`
2. `SmallSerialisedValue`: 0-4k bytes
3. `LargeSerialisedValue`: 4k-64k bytes
4. `SerialisedValue`: a `frequency` of the previous three, where `SmallSerialisedValue` should probably be generated most often

And then we can expose a generator where we can tweak those frequencies

Reply:
Do you really want to have a strict 4k lower bound on `LargeSerialisedValue`, so also ensure we don't shrink to anything smaller? Or just make sure the majority of values is going to be 4k+ bytes?

I'm also not sure we need `TinySerialisedValue` (or do you have a reason in mind?). The potentially useful properties to have that I see are:
- `SmallSerialisedValue`: guaranteed to fit on one page. although I'm not sure this is useful as is since 1. most tests should work for any value size or could use a function to slice only the prefix of an `Entry` and 2. arguably this depends on the key length as well, so maybe it should be `SmallSerialisedKOp`? Are there specific tests we could look at to see what is sensible there?
- `LargeSerialisedValue`: not strictly but usually 4k+, to really make sure we're testing overflow logic. 
- `SerialisedValue`: a good distribution of all sizes with that should work for most tests.

Reply:
> Do you really want to have a strict 4k lower bound on `LargeSerialisedValue`, so also ensure we don't shrink to anything smaller? Or just make sure the majority of values is going to be 4k+ bytes?

0-64k is probably better

> I'm also not sure we need `TinySerialisedValue` (or do you have a reason in mind?). The potentially useful properties to have that I see are:
> 
>     * `SmallSerialisedValue`: guaranteed to fit on one page. although I'm not sure this is useful as is since 1. most tests should work for any value size or could use a function to slice only the prefix of an `Entry` and 2. arguably this depends on the key length as well, so maybe it should be `SmallSerialisedKOp`? Are there specific tests we could look at to see what is sensible there?
> 
>     * `LargeSerialisedValue`: not strictly but usually 4k+, to really make sure we're testing overflow logic.
> 
>     * `SerialisedValue`: a good distribution of all sizes with that should work for most tests.

> Do you really want to have a strict 4k lower bound on `LargeSerialisedValue`, so also ensure we don't shrink to anything smaller? Or just make sure the majority of values is going to be 4k+ bytes?


It depends on the distribution of `SmallSerialisedValue` sizes whether a separate `TinySerialisedValue` is useful. If the distribution is uniform, then tiny is probably useful because on average values will be close to large. One argument for generating tiny values is the dual of the argument for generating values: that there are cases where values being tiny are handled differently. For example, in the page format we only reach interesting cases for the blobref bitmap and op type bitmap when we have >64 or >32 entries in the page respectively

Reply:
Maybe it's an argument for having a number of different distributions, not necessarily for having a `TinySerialisedValue` newtype

Reply:
I moved these changes to #180